### PR TITLE
kuma,kuma-cp,kuma-dp,kuma-experimental,kumactl: 1.8.1 → 2.3.1

### PR DESCRIPTION
**Pull request by kirillrdy**
## Description of changes

https://github.com/kumahq/kuma/blob/master/CHANGELOG.md
https://github.com/kumahq/kuma/compare/1.8.1...2.3.1

also fixes https://github.com/NixOS/nixpkgs/pull/245159

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/cluster/kuma/default.nix [hunk 1]
```diff
@@ -9,27 +9,27 @@
...
 
   src = fetchFromGitHub {
     owner = "kumahq";
     repo = "kuma";
     rev = version;
-    sha256 = "sha256-hNfgiMX3aMb8yjXjFKz73MczOeJyOI3Tna/NRSJBSzs=";
+    sha256 = "sha256-BayfHBTTqgc0ArD6ux9HOqaZy0GrEpqgDa7zHZtiG2I=";
   };
 
+  vendorSha256 = "sha256-St+jGks7ojKrgecmN7UJ9FjGrmjtgEKsunSY+4itUyA=";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/kuma/default.nix:28]: This could use hash instead

File: pkgs/top-level/aliases.nix [hunk 2]
```diff
@@ -844,6 +844,7 @@ mapAliases ({
   kubei = kubeclarity; # Added 2023-05-20
   kubeless = throw "kubeless is deprecated and archived by upstream"; # Added 2022-04-05
   kubicorn = throw "kubicorn has been dropped due to the lack of maintenance from upstream since 2019"; # Added 2022-05-30
+  kuma-prometheus-sd = throw "kuma-prometheus-sd has been depricated upstream"; # Added 2023-07-02
```
Comment (SuperSandro2000) [inline pkgs/top-level/aliases.nix:847]:
```suggestion
  kuma-prometheus-sd = throw "kuma-prometheus-sd has been deprecated upstream"; # Added 2023-07-02
```

File: pkgs/applications/networking/cluster/kuma/default.nix (hunk 1 above)
Comment (kirillrdy) [reply pkgs/applications/networking/cluster/kuma/default.nix]: yep changed to hash and vendorHash

File: pkgs/top-level/aliases.nix (hunk 2 above)
Comment (kirillrdy) [reply pkgs/top-level/aliases.nix]: done

**Comment by kirillrdy** [comment]
@Janik-Haag  this is the last one :pray:  after that we can close https://github.com/NixOS/nixpkgs/pull/245159

File: pkgs/applications/networking/cluster/kuma/default.nix [hunk 3]
```diff
@@ -9,27 +9,27 @@
...
     owner = "kumahq";
     repo = "kuma";
     rev = version;
-    sha256 = "sha256-hNfgiMX3aMb8yjXjFKz73MczOeJyOI3Tna/NRSJBSzs=";
+    hash = "sha256-BayfHBTTqgc0ArD6ux9HOqaZy0GrEpqgDa7zHZtiG2I=";
   };
 
+  vendorHash = "sha256-St+jGks7ojKrgecmN7UJ9FjGrmjtgEKsunSY+4itUyA=";
+
   doCheck = false;
```
Comment (Janik-Haag) [inline pkgs/applications/networking/cluster/kuma/default.nix:30]: can you please add a comment why the test is disabled?

**Comment by Janik-Haag** [comment]
Result of `nixpkgs-review pr 246651` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>kuma</li>
    <li>kuma-cp</li>
    <li>kuma-dp</li>
    <li>kuma-experimental</li>
    <li>kumactl</li>
  </ul>
</details>

File: pkgs/applications/networking/cluster/kuma/default.nix (hunk 3 above)
Comment (kirillrdy) [reply pkgs/applications/networking/cluster/kuma/default.nix:31]: done

**Review by Janik-Haag — APPROVED** [review]
LGTM

**Review by AndersonTorres — APPROVED** [review]

**Comment by Janik-Haag** [comment]
Seems like you have to disable the tests if the build environment is darwin since it throws some errors about dns. After that it should be good to merge.

**Comment by kirillrdy** [comment]
@Janik-Haag build failure on darwin is happening in dependency

will try to fix it here https://github.com/NixOS/nixpkgs/pull/250749

**Pull request merged**